src/app/store.ts

```diff
@@ -11,23 +11,27 @@ import selectedTagColourReducer from "../features/selectedTagColourSlice";
 import createTagMenuDataReducer from "../features/createTagMenuDataSlice";
 import modalActionMenusVisibilityReducer from "../features/modalActionMenusVisibilitySlice";
 
-const store = configureStore({
-  reducer: {
-    board: boardReducer,
-    lists: listsReducer,
-    cards: cardsReducer,
-    boardOptions: boardOptionsReducer,
-    currentSelectedCard: currentSelectedCardReducer,
-    currentSelectedList: currentSelectedListReducer,
-    listOptionsMenuPosition: listOptionsMenuPositionReducer,
-    tags: tagsReducer,
-    selectedTagColour: selectedTagColourReducer,
-    createTagMenuData: createTagMenuDataReducer,
-    modalActionMenusVisibility: modalActionMenusVisibilityReducer,
-  },
+const rootReducer = combineReducers({
+  board: boardReducer,
+  lists: listsReducer,
+  cards: cardsReducer,
+  boardOptions: boardOptionsReducer,
+  currentSelectedCard: currentSelectedCardReducer,
+  currentSelectedList: currentSelectedListReducer,
+  listOptionsMenuPosition: listOptionsMenuPositionReducer,
+  tags: tagsReducer,
+  selectedTagColour: selectedTagColourReducer,
+  createTagMenuData: createTagMenuDataReducer,
+  modalActionMenusVisibility: modalActionMenusVisibilityReducer,
 });
 
-export type RootState = ReturnType<typeof store.getState>;
-export default store;
+export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
+  return configureStore({
+    reducer: rootReducer,
+    preloadedState,
+  });
+};
```

Comment:
I've had to change this slightly so it can be used for tests without duplicating code. Instead of exporting a store, we now export the function to setup a store. This is useful for testing components with Redux bits in them and checking the state is what we expect it to be